### PR TITLE
Add an argument to set bucket_cap_mb for PyTorch DDP

### PR DESCRIPTION
**Pull request by changlan**
# What does this PR do?

`bucket_cap_mb` determines the bucket size that PyTorch's `DistributedDataParallel` will group gradient parameters in. In some occasions, tuning this parameter will have significant impact on the distributed training performance. This PR allows user to change the `bucket_cap_mb` parameter via a new training argument `--ddp_bucket_cap_mb`.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1007,11 +1007,15 @@ def _wrap_model(self, model, training=True):
                 find_unused_parameters = not model.is_gradient_checkpointing
             else:
                 find_unused_parameters = True
+            kwargs = {}
```
Comment (sgugger) [inline src/transformers/trainer.py:1010]: Maybe put the `find_unused_parameters` here and remove it from the wrapper below?

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

File: src/transformers/trainer.py (hunk 1 above)
Comment (changlan) [reply src/transformers/trainer.py]: Sure, done!

**Comment by sgugger** [comment]
There is just the formatting issue to deal with before we can merge. Can you run `make style` on your branch to fix it?
Thanks!

**Comment by changlan** [comment]
Done - thanks for the review!

**Comment by sgugger** [comment]
Mmm, are you sure you installed the proper versions of the formatting tools with `pip install transformers[dev]` or `pip install transformers[quality]` ? The failure is still there.

**Comment by changlan** [comment]
My bad - I forgot to push the latest commit. Should be working now?

**Pull request merged**